apps/vocabulary/main.go

```diff
@@ -0,0 +1,261 @@
+		key_names = append(key_names, key)
+		key_names_lower = append(key_names_lower, strings.ToLower(key))
+	}
+	sort.Strings(key_names)
+	sort.Strings(key_names_lower)
+	f2, ferror := os.Create("IndexAlphaSort.csv")
+	if ferror != nil {
+		fmt.Println(ferror)
+		f2.Close()
+		return
+	}
+	for _, key := range key_names {
+		temp := fmt.Sprintf("%s,%d\n", key, names[key])
+		f2.WriteString(temp)
+
+	}
+	f2.Close()
+
+	f3, ferror := os.Create("IndexAlphaLowerSort.csv")
+	if ferror != nil {
+		fmt.Println(ferror)
+		f3.Close()
+		return
+	}
+	for _, key := range key_names_lower {
+		temp := fmt.Sprintf("%s,%d\n", key, names[key])
+		f3.WriteString(temp)
+
+	}
+	f3.Close()
+
+	f4, ferror := os.Create("IndexRegular.csv")
+	if ferror != nil {
+		fmt.Println(ferror)
+		f4.Close()
+		return
+	}
+	for k, _ := range names {
+		temp := fmt.Sprintf("%s\n", k)
+		f4.WriteString(temp)
+	}
+	f4.Close()
+}
+
+func printDocument(document *pb.Document, schemas map[string]int, operationId map[string]int, names map[string]int, properties map[string]int) {
+	//Start
+	if document.Definitions != nil && document.Definitions.AdditionalProperties != nil {
+		for _, pair := range document.Definitions.AdditionalProperties {
+			addToMap(pair.Name, schemas)
+			printSchema(pair.Value, properties)
+		}
+	}
+	for _, pair := range document.Paths.Path {
+		v := pair.Value
+		if v.Get != nil {
+			printOperation(v.Get, operationId, names)
+		}
+		if v.Post != nil {
+			printOperation(v.Post, operationId, names)
+		}
+		if v.Put != nil {
+			printOperation(v.Put, operationId, names)
+		}
+		if v.Patch != nil {
+			printOperation(v.Patch, operationId, names)
+		}
+		if v.Delete != nil {
+			printOperation(v.Delete, operationId, names)
+		}
+	}
+}
+
+//^^^ Get rid of print post/get/indent
+
+func printOperation(operation *pb.Operation, operationId map[string]int, names map[string]int) {
+	if operation.OperationId != "" {
+		addToMap(operation.OperationId, operationId)
+	}
+	for _, item := range operation.Parameters {
+		switch t := item.Oneof.(type) {
+		case *pb.ParametersItem_Parameter:
+			switch t2 := t.Parameter.Oneof.(type) {
+			case *pb.Parameter_BodyParameter:
+				addToMap(t2.BodyParameter.Name, names)
+			case *pb.Parameter_NonBodyParameter:
+				switch t3 := t2.NonBodyParameter.Oneof.(type) {
+				case *pb.NonBodyParameter_FormDataParameterSubSchema:
+					addToMap(t3.FormDataParameterSubSchema.Name, names)
+				case *pb.NonBodyParameter_HeaderParameterSubSchema:
+					addToMap(t3.HeaderParameterSubSchema.Name, names)
+				case *pb.NonBodyParameter_PathParameterSubSchema:
+					addToMap(t3.PathParameterSubSchema.Name, names)
+				case *pb.NonBodyParameter_QueryParameterSubSchema:
+					addToMap(t3.QueryParameterSubSchema.Name, names)
+				}
+			}
+		}
+	}
+}
+
+func printSchema(schema *pb.Schema, properties map[string]int) {
+	if schema.Properties != nil {
+		for _, pair := range schema.Properties.AdditionalProperties {
+			addToMap(pair.Name, properties)
+		}
+	}
```

Comment:
In Go, we prefer the "special case" logic to be indented and the main logic to be as far to the left as possible.

So in this case we'd invert the `if` condition check, early `return` in the condition body, otherwise fall through to the mainline logic of the `for` loop. 

Related guidance: https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow